src/Umbraco.Web/HealthCheck/Checks/Security/TlsCheck.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Web.HealthCheck.Checks.Security
+{
+    [HealthCheck("6D7D4E11-758E-4697-BA72-7E02A530CBD4",
+        "TLS Check",
+        Description = "Checks that your site is set up to use at least TLS 1.2 for outgoing connections.",
+        Group = "Security")]
+    public class TlsCheck : HealthCheck
+    {
+        private readonly ILocalizedTextService _textService;
+
+        public TlsCheck(HealthCheckContext healthCheckContext) : base(healthCheckContext)
+        {
+            _textService = healthCheckContext.ApplicationContext.Services.TextService;
+        }
+
+        public override IEnumerable<HealthCheckStatus> GetStatus()
+        {
+            return new[] { CheckTls() };
+        }
+
+        public override HealthCheckStatus ExecuteAction(HealthCheckAction action)
+        {
+            throw new InvalidOperationException("TlsCheck has no executable actions");
+        }
+
+        public HealthCheckStatus CheckTls()
+        {
+            bool success = (int)ServicePointManager.SecurityProtocol >= (int)SecurityProtocolType.Tls12;
```

Comment:
This check is incorrect and not future-proof!

Because [`SecurityProtocolType`](https://docs.microsoft.com/en-us/dotnet/api/system.net.securityprotocoltype?view=netframework-4.5#fields) is a flags-enum, if both `Tls11` and `Tls12` are set, it's value (768 + 3072) would be greater than 3072 and return success, while TLS 1.1 is still allowed.

Starting from .NET Framework 4.7, the default value is changed to [`SystemDefault`](https://docs.microsoft.com/en-us/dotnet/api/system.net.servicepointmanager.securityprotocol?view=netframework-4.7.2#remarks), which is the most secure setting, but this check would return otherwise.

Reply:
Umbraco v7 is stuck on .NET 4.5 for now.

Reply:
TLS 1.1 should still be allowed, this check only indicated if it will ATTEMPT to connect on 1.2.

It will not be merged to v8 as it is not needed there.